conversion.go

```diff
@@ -0,0 +1,129 @@
+package safecast
+
+import (
+	"errors"
+	"fmt"
+	"math"
+)
+
+type Integer interface {
+	~int | ~uint | ~int8 | ~uint8 | ~int16 | ~uint16 | ~int32 | ~uint32 | ~int64 | ~uint64
+}
+
+var ErrOutOfRange = errors.New("out of range")
+
+func ToInt[T Integer](i T) (int, error) {
+	if i > 0 && uint64(i) > math.MaxInt {
+		return 0, fmt.Errorf("%w: %d is greater than math.MaxInt", ErrOutOfRange, i)
+	}
+
+	if i < 0 && uint64(-i) > -math.MinInt {
+		return 0, fmt.Errorf("%w: %d is less than math.MinInt", ErrOutOfRange, i)
+	}
+
+	return int(i), nil
+}
+
+func ToUint[T Integer](i T) (uint, error) {
+	if i < 0 {
+		return 0, fmt.Errorf("%w: %d is negative", ErrOutOfRange, i)
+	}
+
+	if uint64(i) > math.MaxUint {
+		return 0, fmt.Errorf("%w: %d is greater than math.MaxUint", ErrOutOfRange, i)
+	}
+
+	return uint(i), nil
+}
+
+func ToInt8[T Integer](i T) (int8, error) {
+	if i > math.MaxInt8 {
+		return 0, fmt.Errorf("%w: %d is greater than math.MaxInt8", ErrOutOfRange, i)
+	}
+
+	if i < 0 && uint64(-i) > -math.MinInt8 {
```

Comment:
yeah this is definitively too ugly
```suggestion
	if i < 0 && int64(i) < math.MinInt8 {
```
and I think might as well just be `i<math.MinInt8` directly (at least for int8 there is no smaller one)